File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/repository/BikeRepository.java
package com.lattis.ellipse.domain.repository;

import com.lattis.ellipse.domain.model.Bike;
import com.lattis.ellipse.domain.model.Ride;
import com.lattis.ellipse.domain.model.SearchBike;

import io.reactivex.Observable;
/**
 * Created by ssd3 on 3/21/17.
 */

public interface BikeRepository {
    Observable<SearchBike> findBikes(double latitude, double longitude );
    Observable<Ride> bookBike(Bike bike,boolean by_scan,double latitude,double longitude);
    Observable<Boolean> cancelBike(int bike_id, boolean bike_damaged, boolean lockIssue);
    Observable<Bike> bikeDetails(int bike_id, int qr_code_id);
    Observable<Boolean> updateBikeMetaData(int bike_id,int bike_battery_level, int lock_battery_level, String firmware_version, Boolean shackle_jam);
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/repository/BikeRepository.kt
package com.lattis.domain.repository

import com.lattis.domain.models.*
import io.reactivex.rxjava3.core.Observable

interface BikeRepository {

    fun searchBike(northEast: Location?,southWest:Location?): Observable<Rentals>
    fun findByQRCode(qr_code:String):Observable<Rental>
    fun bookBike(
        bike: Bike,
        by_scan: Boolean,
        latitude: Double?,
        longitude: Double?,
        device_token:String,
        pricing_option_id:Int?
    ): Observable<Ride>

    fun bikeDetails(bike_id: Int, qr_code_id: Int, iot_qr_code:String?): Observable<Bike>

    fun updateBikeMetaData(
        bike_id: Int,
        bike_battery_level: Int?,
        lock_battery_level: Int?,
        firmware_version: String?,
        shackle_jam: Boolean
    ): Observable<Boolean>

    fun cancelBike(
        bike_id: Int,
        bike_damaged: Boolean,
        lockIssue: Boolean
    ): Observable<Boolean>

    fun lockUnlockIotBike(
        bike_id: Int,
        lock:Boolean,
        controller_key:List<String>?
    ):Observable<IoTBikeLockUnlockCommandStatus>

    fun getIotBikeStatus(
        bike_id: Int,
        controller_key:String?
    ):Observable<IoTBikeStatus>


    fun getLinkaIotBikeStatus(bike_id: Int,commandId:String): Observable<IoTBikeLockUnlockCommandStatus>
    fun searchBikeByName(bike_name:String?):Observable<List<Bike>>

    fun unlockSentinelBike(bike_id:Int):Observable<Boolean>
}